tests/src/Web/WebTestCase.php

```diff
@@ -0,0 +1,234 @@
+    {
+        $this->getRulesRepo()->addRelation($ruleModel);
+    }
+
+    public function getJsonResponse()
+    {
+        /** @var Response $response */
+        $response = $this->getResponse();
+        if (!$response->isOk()) {
+            $this->fail('Response returned was not 200: '.$response->getContent());
+        }
+        $json = json_decode($response->getContent());
+
+        return $json;
+    }
+
+    public function newClient()
+    {
+        return $this->api = static::createClient();
+    }
+
+    public function getResponse()
+    {
+        return $this->api->getResponse();
+    }
+
+    public function createConfiguration()
+    {
+        if (file_exists(__DIR__.'/../../../config/local.php')) {
+            $this->isLocal = true;
+            $config = include __DIR__.'/../../../config/local.php';
+        } else {
+            $this->isLocal = false;
+            $config = include __DIR__.'/../../../config/ci.php';
+        }
+
+        $config['tables']['rules'] = 'Rules__test';
+        $config['tables']['references'] = 'References__test';
+
+        return $this->modifyConfiguration($config);
+    }
+
+    public function getDatabase(): Connection
+    {
+        return $this->kernel->getApp()['db'];
+    }
+
+    public function getRulesRepo(): RuleModelRepository
+    {
+        return $this->kernel->getApp()['rules.repository'];
+    }
+
+    public function modifyConfiguration($config)
+    {
+        return $config;
+    }
+
+    protected function mapHeaders($headers)
+    {
+        $httpHeaders = [];
+        foreach ($headers as $key => $header) {
+            $httpHeaders['HTTP_'.$key] = $headers[$key];
+        }
+
+        return $httpHeaders;
+    }
+
+    protected function jsonRequest(string $verb, string $endpoint, array $params = array(), array $headers = array())
+    {
+        $server = array_merge(array(
+            'HTTP_ACCEPT' => 'application/json',
+            'CONTENT_TYPE' => 'application/json',
+        ), $this->mapHeaders($headers));
+
+        return $this->api->request(
+            $verb,
+            $endpoint,
+            $params,
+            [],
+            $server
+        );
+    }
+
+    /**
+     * Creates the application.
+     *
+     * @return HttpKernelInterface
+     */
+    public function createApplication()
+    {
+        $this->kernel = new Kernel($this->createConfiguration(), false);
+
+        return $this->kernel->getApp();
+    }
+
+    public function setUp()
+    {
+        parent::setUp();
+        $lines = $this->runCommand('generate:database');
+//        $this->assertStringStartsWith('Database created successfully.', $lines[0], 'Failed to run test during set up');
+    }
+
+    public function tearDown()
+    {
+        $this->getDatabase()->query(sprintf('SET FOREIGN_KEY_CHECKS=%s', (int) false));
+        // Delete database.
+        $this->getDatabase()->getSchemaManager()->dropTable('References__test');
```

Comment:
could be optimized to only delete data rather than dropping the tables, if the tests take a long time to run